invenio_pidstore/providers/recordid.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Record ID provider."""
+
+from __future__ import absolute_import, print_function
+
+from ..provider import LocalPidProvider
+
+
+class RecordID(LocalPidProvider):
+    """Provider for recids."""
+
+    pid_type = 'recid'
+
+    @classmethod
+    def is_provider_for_pid(cls, pid_str):
+        """A recid is valid is it is ``None`` or ``Integer``."""
+        return isinstance(pid_str, int) or pid_str.isdigit()
```

Comment:
The documentation seems to say that `pid_str` can be set to `None`, which would make this line fail with `AttributeError: 'NoneType' object has no attribute 'isdigit'`.